library/headers.inc.php

```diff
@@ -42,43 +46,66 @@ function include_css_library($path)
-    <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path'].$v ; ?>/index.js"></script>    
-    <?php 
-    } }
+    /* First checking for any library in a directory matching the string "/jquery-min-/".
+     * When one is found, use that value in the URL string, then add "index.js"
+     * In order to decrease the chance of including and library calling problems, some measures were taken:
+     * - The array in the argument is completely converted to lower-case, meaning that there is not any sensitivity between jQuery & jquery or fancyBox & fancybox
+     * - Some libraries can be called with or without "-". For example: "font-awesome" and "fontawesome" => those 2 are the same thing
+     */
+    
+    $library_array = array_map('strtolower', $library_array);
+
+    foreach ($library_array as $v){ 
+        if (preg_match("/jquery-min-/", $v)) {?>
+            <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path'].$v ; ?>/index.js"></script>    
+        <?php 
+        }
+    }   
         
-    if (in_array("bootstrap",$library_array)){   ?>
+    if (in_array("bootstrap", $library_array)) {   ?>
         <link rel="stylesheet" href="<?php echo $GLOBALS['standard_js_path']; ?>bootstrap-3-3-4/dist/css/bootstrap.min.css" type="text/css">
         <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path']; ?>bootstrap-3-3-4/dist/js/bootstrap.min.js"></script>
     <?php
     }
-    if (in_array("fancybox",$library_array)){   ?>
+
+    if (in_array("fancybox", $library_array)) {   ?>
         <link rel="stylesheet" href="<?php echo $GLOBALS['css_path']; ?>fancybox/jquery.fancybox-1.2.6.css" media="screen" />
         <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path']; ?>fancybox/jquery.fancybox-1.2.6.js"></script>
     <?php
     }
-    if (in_array("knockout",$library_array)){   ?>
+
+    if (in_array("knockout", $library_array)) {   ?>
         <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path']; ?>knockout/knockout-3.4.0.js"></script>
     <?php
     }
-    if (in_array("datepicker",$library_array)){   ?>
+
+    if (in_array("datepicker", $library_array)) {   ?>
         <link rel="stylesheet" href="<?php echo $GLOBALS['css_path']; ?>jquery-datetimepicker/jquery.datetimepicker.css" media="screen" />
         <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path']; ?>jquery-datetimepicker/jquery.datetimepicker.full.min.js"></script>
     <?php
     }
-    if (in_array('font-awesome', $library_array)) {?>
+
+    if (in_array('font-awesome', $library_array) || in_array('fontawesome', $library_array)) {   ?>
         <link rel="stylesheet" href="<?php echo $GLOBALS['assets'] ?>/fonts/font-awesome-4-6-3/css/font-awesome.min.css" type="text/css">
     <?php
     }
-    if (in_array("jquery-ui",$library_array)){   ?>
+
+    if (in_array("jquery-ui", $library_array) || in_array("jqueryui", $library_array)) {   ?>
```

Comment:
Reason we have or might have an inconsistency?

Reply:
I wanted to do this to prevent possible future issues when calling libraries (not to fix any inconsistency)